cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectDockerfile.java

```diff
@@ -154,13 +154,35 @@ private void connectorPluginsPreStage(PrintWriter writer, List<Plugin> plugins)
 
-                    Cmd cmd = run("curl", "-f", "-L", "--create-dirs", "--output", "/tmp/" + artifactDir + "/pom.xml", assembleResourceUrl(repo, mvn, "pom"))
-                            .andRun("echo", settingsXml).redirectTo(settingsFile) // Create the settings file
-                            .andRun("mvn", "dependency:copy-dependencies", "-s", settingsFile,
-                                    "-DoutputDirectory=/tmp/artifacts/" + artifactDir, "-f", "/tmp/" + artifactDir + "/pom.xml")
-                            .andRun("curl", "-f", "-L", "--create-dirs", "--output",
-                                    "/tmp/artifacts/" + artifactDir + "/" + mvn.getArtifact() + "-" + mvn.getVersion() + ".jar",
-                                    assembleResourceUrl(repo, mvn, "jar"));
+                    Cmd cmd;
+                    if (Boolean.TRUE.equals(mvn.getInsecure()))    {
+                        // Insecure download => disables TLS certificate checks in curl and Maven commands
+                        cmd = run("curl", "-f", "-k", "-L", "--create-dirs", "--output", "/tmp/" + artifactDir + "/pom.xml", assembleResourceUrl(repo, mvn, "pom"))
+                                .andRun("echo", settingsXml).redirectTo(settingsFile) // Create the settings file
+                                .andRun("mvn", "dependency:copy-dependencies", "-s", settingsFile,
+                                        "-DoutputDirectory=/tmp/artifacts/" + artifactDir, "-Daether.connector.https.securityMode=insecure",
+                                        "-Dmaven.wagon.http.ssl.insecure=true", "-Dmaven.wagon.http.ssl.allowall=true",
+                                        "-Dmaven.wagon.http.ssl.ignore.validity.dates=true", "-f", "/tmp/" + artifactDir + "/pom.xml")
+                                .andRun("curl", "-f", "-k", "-L", "--create-dirs", "--output",
+                                        "/tmp/artifacts/" + artifactDir + "/" + mvn.getArtifact() + "-" + mvn.getVersion() + ".jar",
+                                        assembleResourceUrl(repo, mvn, "jar"));
+                    } else {
+                        cmd = run("curl", "-f", "-L", "--create-dirs", "--output", "/tmp/" + artifactDir + "/pom.xml", assembleResourceUrl(repo, mvn, "pom"))
+                                .andRun("echo", settingsXml).redirectTo(settingsFile) // Create the settings file
+                                .andRun("mvn", "dependency:copy-dependencies", "-s", settingsFile,
+                                        "-DoutputDirectory=/tmp/artifacts/" + artifactDir, "-f", "/tmp/" + artifactDir + "/pom.xml")
+                                .andRun("curl", "-f", "-L", "--create-dirs", "--output",
+                                        "/tmp/artifacts/" + artifactDir + "/" + mvn.getArtifact() + "-" + mvn.getVersion() + ".jar",
+                                        assembleResourceUrl(repo, mvn, "jar"));
+                    }
```

Comment:
it's a pity that it's not that simple to factor out some common code here, considering that most of the arguments are actually the same. I see that `run` has a String varargs but anyway building a list of String(s) with arguments and then call the `run` just one time could be not that simple. Not saying you should do that, anyway.

Reply:
I tried to have originally the code only once. But it is quite complicated with how the escaping has to be done etc. So at the end I decided it is better to just have it twice like that because any attempt to have the command built only once were always much worse (at least in my opinion) than this :-(.